packages/react-ui-components/src/components/CodeEditor/types.ts

```diff
@@ -1,8 +1,29 @@
 import type { editor } from 'monaco-editor';
+import { FunctionComponent } from 'react';
 
 export type OnEditorChangeCallback = (value: string | undefined) => void;
 export type CustomThemeData = editor.IStandaloneThemeData;
-export type CustomEditorOptions = editor.IEditorOptions & editor.IGlobalEditorOptions;
+
+export type KvEditorPaddingOptions = {
+	top?: number;
+	bottom?: number;
+};
+
+export type KvEditorOptions = {
+	/** Use this property to set the editor to a read only mode where the user cannot modify the value. */
+	readOnly?: boolean;
+	/** Use this property to add some padding (in px) at the top of the editor */
+	padding?: KvEditorPaddingOptions;
+};
+
+type ExcludedEditorOptions = 'fontSize' | 'fontFamily';
```

Comment:
Since the type defined by monaco is `string | undefined` for `fontFamily` and `number | undefined` for `fontSize` I've decided to override these props to avoid undefined checks when passing these props around (for instance, to `useFontsApi`), if you disagree on this approach, let me know

Reply:
if we turn all monaco options available from component we can't require the user to set a fontFamily or size.

Reply:
I'm merging our defaults with the `customOptions` the parent provides, so these 2 properties will always be defined.

Depending on the outcome of [this](https://github.com/kelvininc/ui-components/pull/277#discussion_r1157269263) conversation the approach here must be different.

Reply:
I'm going to change this to expose all props as that seems to be the consensus, thanks!